src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -163,4 +171,17 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
             s"from available source columns '${dfColumnNames.mkString(",")}'")
     }
   }
+
+  private def internalColumns(df: DataFrame, indexConfigColumns: Seq[String]): Seq[String] = {
+    val partitionSchema = df.queryExecution.optimizedPlan.collect {
+      case LogicalRelation(HadoopFsRelation(_, pSchema, _, _, _, _), _, _, _) => pSchema
+    }
+
+    val spark = df.sparkSession
+    partitionSchema.head
+      .map(_.name)
+      .filter(ResolverUtils.resolve(spark, _, indexConfigColumns).isEmpty)
+  }
+
+  private val fileName = udf((fullFilePath: String) => FilenameUtils.getBaseName(fullFilePath))
```

Comment:
Question. I guess we wouldn't need the path the file ever right?

Reply:
Which path do you mean? This function is an auxiliary that extract file name from a full path. It makes no assumption about its input args origin or format.